client/src/services/notificationManager.ts

```diff
@@ -131,28 +133,33 @@ export class NotificationManager {
     }
   }
 
-  async addToList(notification: Notification): Promise<void> {
-    this.notifications.push(notification);
+  addToList(notification: Notification): void {
+    this.notifications.value.unshift(notification);
   }
 
   markAsRead(id: string, read = true): void {
-    const notification = this.notifications.find((n) => n.id === id);
+    const notification = this.notifications.value.find((n) => n.id === id);
 
     if (notification) {
       notification.read = read;
     }
+    this.notifications.value = [...this.notifications.value];
   }
 
   getNotifications(): Notification[] {
-    return this.notifications;
+    return this.notifications.value;
   }
 
   hasUnreadNotifications(): boolean {
-    return this.notifications.some((n) => n.read === false);
+    return this.notifications.value.some((n) => n.read === false);
+  }
+
+  getUnreadCount(): number {
```

Comment:
Why not simply return `Ref<number>`? It would avoid adding a computed in the popover.

Reply:
I did that change at the end, forgetting about this, yeah good idea

Reply:
done